api/service/version_image_service_test.go

```diff
@@ -0,0 +1,140 @@
+package service
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/caraml-dev/merlin/mlp"
+	"github.com/caraml-dev/merlin/models"
+	"github.com/caraml-dev/merlin/pkg/imagebuilder/mocks"
+	"github.com/stretchr/testify/mock"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func Test_versionImageService_GetImage(t *testing.T) {
+	type args struct {
+		ctx     context.Context
+		model   *models.Model
+		version *models.Version
+	}
+	tests := []struct {
+		name         string
+		imageBuilder func() *mocks.ImageBuilder
+		args         args
+		want         models.VersionImage
+		wantErr      bool
+	}{
+		{
+			name: "success",
```

Comment:
Hmm should we also include some additional simple test cases to ensure that the correct image builder (or no image builder) gets returned when `s.getImageBuilder(model.Type)` is called? This might mean having to set up two different mocked image builders and then checking that `GetVersionImage` gets called on the right mocked image builder 🤔 

Another test case I can think of is when `imageBuilder.GetVersionImage(ctx, model.Project, model, version)` returns a image that is still being built.